### ISSUE TITLE
FutureWarning and UserWarning 

### ISSUE THREAD

**Issue by cheuS1-n** [victorca25/traiNNer#73]
D:\traiNNer\codes\models\base_model.py:921: FutureWarning: Non-finite norm encountered in torch.nn.utils.clip_grad_norm_; continuing anyway. Note that the default behavior will change in a future release to error out if a non-finite total norm is encountered. At that point, setting error_if_nonfinite=false will be required to retain the old behavior.
  self.grad_clip(
C:\Python39\lib\site-packages\torch\optim\lr_scheduler.py:129: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
  
 How fix this?

**Comment by victorca25** [comment]
Hello!

> D:\traiNNer\codes\models\base_model.py:921: FutureWarning: Non-finite norm encountered in torch.nn.utils.clip_grad_norm_; continuing anyway. Note that the default behavior will change in a future release to error out if a non-finite total norm is encountered. At that point, setting error_if_nonfinite=false will be required to retain the old behavior.

This means the model is probably unstable and some values are becoming InF. Normally gradient clipping will set those values to max at the defined values, but looks like this behavior is changing. You can try lowering the learning rate a bit to see if the error goes away, otherwise, there might be some other configuration causing these values to become infinite.

> C:\Python39\lib\site-packages\torch\optim\lr_scheduler.py:129: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`. Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
> warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "

This is normal when using autocasting, no problem.

**Comment by cheuS1-n** [comment]
Thanks for the answer! But I still have a problem, the program does not use video card, only video memory and and one epoch lasts for a very long time.
Core of video card dont used or used non correct, but only the processor. (G.Translate)
![Знімок екрана 2022-12-13 093533](https://user-images.githubusercontent.com/50949667/207254841-e0c8277a-e658-4802-bcb0-d8dd288f4185.png)

**Comment by victorca25** [comment]
How long is an epoch taking? Regarding the GPU utilization in Windows task manager, please check: https://github.com/victorca25/traiNNer/issues/72

**Comment by cheuS1-n** [comment]
The epoch lasts 250-400 seconds(I don't know whether it is long but I think for a long time).
I load windows task manager screen, and GPU-Z.
![image](https://user-images.githubusercontent.com/50949667/207257490-861fa397-3fab-49b0-96ca-445387905314.png)
![image](https://user-images.githubusercontent.com/50949667/207258152-11a36207-747c-4f6e-a77c-5a2fdb6aabe3.png)

Please tell us the video card is normally used? Because the uneven graphic is surprising. (Results from GPU Z and from WIndows Task Manager very different)

**Comment by victorca25** [comment]
How many images are you using for training? The time to complete a full epoch depends on how many images you have. More concretely, one epoch means that all the images have been used and it has to load all the images again with the dataloader.

**Comment by cheuS1-n** [comment]
I have ±2000 2K resolution images.
